From bc6f273a0475a1fa7ab56bc1e498ee62c96aa660 Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Wed, 8 Apr 2020 16:10:38 +0200
Subject: [PATCH] units: drop dependency on systemd-remount-fs.service from
 systemd-pstore.service

This dependency is now generated automatically given we use
StateDirectory=. Moreover the combination of Wants= and After= was too
strong anway, as whether remount-fs is pulled in or not should not be up
to systemd-pstore.service, and in fact is part of the initial
transaction anyway.

[dtardon: This only removes Wants=, not After=, because I haven't
backported the auto-generation code the description talks about. The
code is simple, but it's just an optimisation allowing for slightly
shorter unit files, hence I don't think we really need it.]

(cherry picked from commit 0c978faa16fa9ecf92f0bbb5c7cc709dc472d115)

Related: #2158832
---
 units/systemd-pstore.service.in | 1 -
 1 file changed, 1 deletion(-)

diff --git a/units/systemd-pstore.service.in b/units/systemd-pstore.service.in
index 37fcf878f0..9a86f3145c 100644
--- a/units/systemd-pstore.service.in
+++ b/units/systemd-pstore.service.in
@@ -13,7 +13,6 @@ Documentation=man:systemd-pstore(8)
 ConditionDirectoryNotEmpty=/sys/fs/pstore
 ConditionVirtualization=!container
 DefaultDependencies=no
-Wants=systemd-remount-fs.service
 After=systemd-remount-fs.service
 Before=sysinit.target
 
